### PR TITLE
chore(ci): prepush hook checks for a conventional commit on the branch

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** NA

## Summary
- You are allowed to push if there is at least one conventional commit on a branch
- pushes to master are exempt, but they still prompt to make sure its on purpose (version releases are non conventional commits)

### CODE REVIEW

**Comment by benelan** [comment]
This actually isn't ready to install I could use some help @jcfranco. The regex I have in this PR isn't great, the scope can't have dashes or commas. I tried a bunch of other stuff that works in online editors but not in the bash script, eg https://regex101.com/r/1kURw0/1

Bash uses Extended Regular Expression syntax right? Or am I missing something

File: package.json [hunk 1]
```diff
@@ -50,7 +50,8 @@
     "util:prep-next-from-existing-build": "ts-node --project ./tsconfig-node-scripts.json support/prepReleaseCommit.ts --next",
     "util:publish-next": "npm publish --tag next",
     "util:push-tags": "git push --follow-tags",
-    "util:run-tests": "npm run util:copy-icons && stencil test --no-docs --spec --e2e"
+    "util:run-tests": "npm run util:copy-icons && stencil test --no-docs --spec --e2e",
+    "util:conventional-commit": "ts-node --project ./tsconfig-node-scripts.json support/matchConventionalCommit.ts"
```
Comment (jcfranco) [inline package.json:54]: Nitpick: place in alphabetical order

**Review by jcfranco — APPROVED** [review]
Looks great! Can the util and script be renamed to indicate that it only checks/runs on single-commit branches to ensure a conventional commit message when squash-merged? Now that's a mouthful. 😂

**Comment by benelan** [comment]
> Can the util and script be renamed to indicate that it only checks/runs on single-commit branches to ensure a conventional commit message when squash-merged? Now that's a mouthful. 😂

what about something like `isOneCommitBranchConventional.ts` and `util:is-one-commit-branch-conventional`?

**Comment by jcfranco** [comment]
> Edit: Something like isBranchSquashMergeable might be better because it also passes if there are 2+ commits so its not specific to single-commit branches.

I like where the name is going. One more suggestion: `checkSquashMergeableBranch`?

**Pull request merged**